### PR TITLE
Guide for DB interceptor plugins

### PR DESCRIPTION
**Pull request by igpetrov**
## Description

Related to epic: https://github.com/camunda/product-hub/issues/563

## When should this change go live?

The PR should be merged after this PR: https://github.com/camunda/camunda/pull/20794 which we anticipate to happen during the week of August 19th. We're targeting `8.6.0-alpha5` release.

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [x] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR): will be merged after https://github.com/camunda/camunda/pull/20794 approved.
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

⚠️ The feature will be modified in the nearest future due to architecture streamlining changes. Therefore I have included a warning note in the documentation itself.

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/self-managed/operational-guides/configure-db-custom-headers.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+{label: 'Gradle dependency', value: 'gradle' }
+]
+}>
+
+<TabItem value='maven'>
+
+```xml
+<dependency>
+  <groupId>io.camunda</groupId>
+  <artifactId>camunda-search-plugins</artifactId>
```
Comment (felix-mueller) [inline docs/self-managed/operational-guides/configure-db-custom-headers.md:37]: I found it confusing at first that it's "search" plugin. Do you refer to OpenSearch and Elastic or why search? Isn't it more about the database? camunda-db-plugins e.g.?

**Comment by felix-mueller** [comment]
generally lgtm, pre-approving, just one comment
one question in addition: is this really an operational guide? isn't it more about advanced config / installation? 
@theburi FYI

**Review by felix-mueller — APPROVED** [review]

**Review by conceptualshark — DISMISSED** [review]
I restructured this page and made some edits, and will pre-approve if these changes look good to you. 👍 I also agree with @felix-mueller that this should live in the config/install guides at the moment.

**Comment by igpetrov** [comment]
@felix-mueller @conceptualshark 

Fair point, and that was my thought as well and the only alternative I considered was `Self-Managed/Guides/Configure custom headers`.

**Comment by conceptualshark** [comment]
@igpetrov Is this ready for another review, or still in progress?

**Comment by igpetrov** [comment]
Hi @conceptualshark 

Please, give me few more moments, I'll need to move the page and then we can merge.

**Review by conceptualshark — APPROVED** [review]

**Review by felix-mueller — APPROVED** [review]

File: docs/self-managed/setup/guides/configure-db-custom-headers.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+id: configure-db-custom-headers
+title: "Configure custom headers"
+sidebar_label: "Configure custom headers"
+description: "Learn how to configure DB client custom headers"
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+Custom HTTP headers can be added to a component's Elasticsearch and OpenSearch HTTP clients by creating a new Java plugin, and adding the plugin to your Camunda 8 Self-Managed installation. Using custom HTTP headers may be helpful for adding authentication, tracking, or debugging to your database requests.
```
Comment (felix-mueller) [inline docs/self-managed/setup/guides/configure-db-custom-headers.md:11]: Do we need to add for now that it's not supported for Optimize???

**Review by felix-mueller — APPROVED** [review]

File: docs/self-managed/setup/guides/configure-db-custom-headers.md (hunk 2 above)
Comment (igpetrov) [reply docs/self-managed/setup/guides/configure-db-custom-headers.md:11]: I was expecting for onboarding Optimize before 8.6. For now, we do not include Optimize config value. @felix-mueller do you think, we should explicitly mention Optimize not included?

**Comment by felix-mueller** [comment]
Lets merge then without. Thanks

Am 27.08.2024 18:34 schrieb Igor Petrov ***@***.***>:

@igpetrov commented on this pull request.

________________________________

In docs/self-managed/setup/guides/configure-db-custom-headers.md<https://github.com/camunda/camunda-docs/pull/4171#discussion_r1733205389>:

> @@ -0,0 +1,215 @@
+---
+id: configure-db-custom-headers
+title: "Configure custom headers"
+sidebar_label: "Configure custom headers"
+description: "Learn how to configure DB client custom headers"
+---
+
+import Tabs from ***@***.***/Tabs";
+import TabItem from ***@***.***/TabItem";
+
+Custom HTTP headers can be added to a component's Elasticsearch and OpenSearch HTTP clients by creating a new Java plugin, and adding the plugin to your Camunda 8 Self-Managed installation. Using custom HTTP headers may be helpful for adding authentication, tracking, or debugging to your database requests.


I was expecting for onboarding Optimize before 8.6. For now, we do not include Optimize config value. @felix-mueller<https://github.com/felix-mueller> do you think, we should explicitly mention Optimize not included?

**Pull request merged**